### PR TITLE
Resolve slack notation in solve command to username

### LINKED ISSUE #24: !solve with @<username> syntax stores slack ID, not username
When running the command : 
```
!solved web300 @kileak
```

The bot will store the slack IDs : 
```
@here *web300* : corb3nik has solved the "web300" challenge (together with @u415mede0).
```

### PR DESCRIPTION
**Pull request by Kileak**
This change will allow solving a challenge with specifying support members in slack notation

`!solve challenge @other_user`

Fixes #24

### CODE REVIEW

File: handlers/challenge_handler.py [hunk 1]
```diff
@@ -338,10 +338,17 @@ def execute(self, slack_wrapper, args, channel_id, user_id):
         solver_list = [member['user']['name']]
 
         # Find additional members to add
-        for addSolve in additional_args:
-            if not addSolve in solver_list:
-                solver_list.append(addSolve)
-                additional_solver.append(addSolve)
+        for add_solve in additional_args:
+            if (add_solve.startswith("<@")) and (add_solve.endswith(">")):
```
Comment (Grazfather) [inline handlers/challenge_handler.py]: get rid of extra parens

File: handlers/challenge_handler.py [hunk 2]
```diff
@@ -338,10 +338,17 @@ def execute(self, slack_wrapper, args, channel_id, user_id):
...
+                # slack @notation (resolve to real username)
+                add_solve = add_solve[2:-1].upper()
+                user_obj = slack_wrapper.get_member(add_solve)
+                if (user_obj['ok']):
```
Comment (Grazfather) [inline handlers/challenge_handler.py]: parens

File: handlers/challenge_handler.py [hunk 3]
```diff
@@ -338,10 +338,17 @@ def execute(self, slack_wrapper, args, channel_id, user_id):
...
+                    add_solve = user_obj['user']['name']
+
+            if not add_solve in solver_list:
```
Comment (Grazfather) [inline handlers/challenge_handler.py]: Preferred: `if add_solve not in solver_list`

Comment (Kileak) [reply handlers/challenge_handler.py]: Though I truly cannot understand, how one would prefer this over parenthesis, which makes it directly obvious, which conditions belong together, without having to think a second about operator precedence (and thus improve code readability), it's done...

Comment (Grazfather) [reply handlers/challenge_handler.py]: Python makes efforts to make the english grammatical sense the proper sense. `x not in y` is an operator with the same semantics as `not (x in y)`. You're just used to a language that makes it confusing.

**Comment by Grazfather** [comment]
LGTM

File: handlers/challenge_handler.py [hunk 4]
```diff
@@ -338,10 +338,17 @@ def execute(self, slack_wrapper, args, channel_id, user_id):
         solver_list = [member['user']['name']]
 
         # Find additional members to add
-        for addSolve in additional_args:
-            if not addSolve in solver_list:
-                solver_list.append(addSolve)
-                additional_solver.append(addSolve)
+        for add_solve in additional_args:
+            if add_solve.startswith("<@") and add_solve.endswith(">"):
```
Comment (Grazfather) [inline handlers/challenge_handler.py]: Let's merge this after #79 and use the utility

**Comment by Grazfather** [comment]
Thanks for the rebase.

**Pull request merged**